pandoc-slave/pandoc-workflow/pandocrunner.h

```diff
@@ -0,0 +1,68 @@
+#ifndef PANDOCRUNNER_H
+#define PANDOCRUNNER_H
+
+#include <QObject>
+#include <QString>
+#include <QProcess>
+#include <QMetaEnum>
+
+/*!
+ * \brief The PandocRunner class provides functionality
+ * to run pandoc utility with specified parameters
+ *
+ * Usage:
+ * \code{.cpp}
+ * PandocRunner runner;                         // Create instance runner
+ * QString output = runner.run(                 // Method run() return output content or error
+ *                      PandocRunner::markdown, // Input format
+ *                      PandocRunner::html,     // Output format
+ *                      "path/to/input/file.md" // Path to input file
+ *                  );
+ * if (runner.statusCode != 0)
+ * {
+ *      qDebug() << "ERROR:" << output;         // In this case content holds error
+ * }
+ * else
+ * {
+ *      qDebug() << output;                    // content holds converted string
+ * }
+ * \endcode
+ */
+
+class PandocRunner : public QProcess
+{
+    Q_OBJECT
+
+public:
+    enum PandocFormat {
+        HTML,
+        Markdown,
+        Json,
+        Markdown_Github
+    };
+    Q_ENUM(PandocFormat)
+
+    PandocRunner(QString &pandocExePath, QObject* parent = 0);
+    int statusCode() const;
+    QString content();
+    QString error();
+    QString run(PandocFormat from, PandocFormat to, const QString &file);
```

Comment:
PandocRunner should also have a method that will allow it to grab all conversion results from the pandoc stdout. This will give us a way to convert document and use conversion results immediately without creating any additional files.